### ISSUE TITLE
(Question) Sunshine support +suggestions

### ISSUE THREAD

**Issue by Phantom2332** [schmaldeo/DS4Windows#24]
Currently I'm using a fork of Sunshine called Apollo/Artemis, and the dev of that project is very supportive and keep improving the original sunshine on a daily basis.

Current support of DS4windows with Moonlight is somehow erratic, sometimes more controllers are shown in the computer after using one + hidhide,or the physical DS4 that I use with the computer takes the profile from the virtual sunshine controller I have tried almost everything so I don't know if it's only my issue 

I'm wondering if collaboration between the two projects could improve this issue in anyway.

I'm also very happy with the current updates that we are getting with @schmaldeo Ty so much with your work, maybe you should start a discord server and rename the fork since you are including a lot of new features,and I guess the majority of the people doesn't know about this amazing fork,it is just a suggestion 😀

**Comment by schmaldeo** [comment]
The main issue related to Sunshine's virtual controller support is in how controllers are detected in the program.

Basically, the detection list is filtered using a function that checks whether or not a controller is virtual. This is exactly where the problem is. This needs to be in place for the program not to detect 8 controllers while only a singular one is physically connected. 

However, this behaviour of detecting mutliple virtual controllers seems inconsistent across different machines (and I don't know the reason for it if I'm honest). E.g. on my laptop this is not an issue, but on my PC it is. That's why current Sunshine support is so dodgy. Basic support (the one that you get when you just select `Moonlight Support` in settings) basically skips the check for the controller being virtual if a detected controller has a matching product and vendor ID (as Sunshine emulates DS4, the check is also skipped for physical DS4). It's OK if you haven't got the issue of multiple controllers being detected. 

I tried finding something unique about Sunshine's virtual controller that I could use in the checks, but couldn't come up with anything, so I just added the cooldown-based solution that helps you avoid the multiple detection issue, but it's dodgy due to some of the tradeoffs [described in docs](https://github.com/schmaldeo/DS4Windows/blob/master/doc/moonlight_support.md).

Maybe adding an option to emulate a controller with a unique USB signature to Sunshine would be a good step towards resolving the issue, however I think that this alone would not fix it as if I remember correctly multiple detections were an issue even with virtual controllers. If you could provide me with a link to the fork that you're using I could maybe contact the maintainer or make a PR myself.

To answer some of the suggestions, I don't see a good usecase for a Discord server, if someone wants to report something, they can just create an issue in this repo or go to the [DS4Windows subreddit](https://www.reddit.com/r/DS4Windows/), where people like @eVenent are always very eager to help. 

Renaming a fork is not something I'd do for a few reasons. Firstly, it's different when you create a fork of a project that's still maintained and you're trying to make it a separate project rather than having it just to make PRs. How I see this project is I just took over maintenance from Ryochan7, so renaming it would probably even harm the popularity by using a new name rather than an established one.

For majority of people not knowing about this fork, I don't really know where else I could advertise it. I put up a post about it on the subreddit, I'm getting stars pretty much every day, I think I will just mostly rely on people sharing the project with their friends + maybe GitHub showing it on people's feed.

**Comment by Phantom2332** [comment]
Ty for the detailed answer,yh I also thought about sunshine creating a virtual controller with an unique id,I will ask the dev to do this,to try to improve the situation bit

Keep up the good work 😎

**Comment by schmaldeo** [comment]
Nah what it's doing is it's spoofing vendor and product ID of X360/DS4 depending on what you choose in the settings.

**Comment by Phantom2332** [comment]
Yh but I think that it changes the id (at least the Mac) each boot up,I don't know if that would help with the current config,maybe detecting a special configurable ID in sunshine if thats posible?

**Comment by schmaldeo** [comment]
No I don't think configurable PID is a good idea, because you would also have to enter it in DS4Windows + it adds a lot of complexity for end users that might not be quite familiar with these things.

Better answer would probably be a fixed and unique (or at least one that's not currently supported in DS4W) PID.

**Comment by Phantom2332** [comment]
Any news on this? I commented on the repo of the developer in case it helps

**Comment by schmaldeo** [comment]
https://github.com/ClassicOldSong/Apollo/issues/56#issuecomment-2381632049
> I actually wanted to solve this issue with another method, which should work nice with HidHide and DS4Windows, but I don't have that much time to work on everything, so it might not be feasiable in the near future.

Unless an ingenius idea for it pops up in my head one day (possible but unlikely) or they solve it their way, sadly I don't think I've got anything new to offer.

**Comment by Phantom2332** [comment]
No probs already enjoying the effort and great features that you are including! 😊

**Comment by schmaldeo** [comment]
Also - regarding translations:
![](https://i.imgur.com/7q7Bant.png)

Are you sure it's not some misconfiguration on your end? You can create a separate issue if you include some more details.

**Comment by Phantom2332** [comment]
Will try with a clean install gimme 10 mins

**Comment by schmaldeo** [comment]
There is a possibility that the installation you had before didn't have the localisation dependencies properly linked in the .deps.json file. Back in the early days of this fork I have accidentally shipped a few versions where that was the case so the app would only work in English. If you downloaded one of these and then just used the updater to update your app rather than manually downloading new releases that might have remained the case.

**Comment by Phantom2332** [comment]
Definitely, i used the updater so i guesss that was the issue,ty for the explanation
